### PR TITLE
metrics(processor): Add indexer-like metrics to generic metrics processor

### PR DESCRIPTION
**Pull request by ayirr7**

### CODE REVIEW

File: rust_snuba/src/processors/generic_metrics.rs [hunk 1]
```diff
@@ -193,6 +194,9 @@ impl Parse for CountersRawRow {
         let timeseries_id =
             generate_timeseries_id(from.org_id, from.project_id, from.metric_id, &from.tags);
         let (tag_keys, tag_values): (Vec<_>, Vec<_>) = from.tags.into_iter().unzip();
+
+        metric!(Gauge: "process_messages.messages.tags_len", tag_keys.len(), "metric_type" => "counter");
```
Comment (untitaker) [inline rust_snuba/src/processors/generic_metrics.rs:198]: you should be using a timer for all of those tags_len metrics IMO

File: rust_snuba/src/processors/generic_metrics.rs [hunk 2]
```diff
@@ -242,6 +246,8 @@ where
 
     let result: Result<Option<T>, anyhow::Error> = T::parse(msg, config);
 
+    metric!(Timer: "process_messages.messages.size", payload_bytes.len());
```
Comment (untitaker) [inline rust_snuba/src/processors/generic_metrics.rs:249]: I think the prefix `process_messages.messages` is not necessary, that function doesn't exist here. Please go with something more specific to generic-metrics. `generic_metrics.message.size` seems better

**Review by untitaker — APPROVED** [review]

File: rust_snuba/src/processors/generic_metrics.rs [hunk 3]
```diff
@@ -289,9 +295,15 @@ impl Parse for SetsRawRow {
             MetricValue::Set(values) => values.into_vec(),
             _ => return Ok(Option::None),
         };
+
+        metric!(Timer: "process_messages.messages.sets_value_len", sets_values.len());
```
[the remark this answers is not in the document]
Comment (untitaker) [reply rust_snuba/src/processors/generic_metrics.rs:299]: example here: https://github.com/getsentry/arroyo/blob/51addf8851d4dda7247843464dbf6441ca9aaa3a/rust-arroyo/src/processing/mod.rs#L175

File: rust_snuba/src/processors/generic_metrics.rs (hunk 1 above)
Comment (ayirr7) [reply rust_snuba/src/processors/generic_metrics.rs:198]: fixed

File: rust_snuba/src/processors/generic_metrics.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ayirr7) [reply rust_snuba/src/processors/generic_metrics.rs:299]: got it, thanks. i was confused by the macro, maybe i should give it a closer look to understand it

[the remark this answers is not in the document]
Comment (ayirr7) [reply rust_snuba/src/processors/generic_metrics.rs:299]: though, one question, can we use metric tags in the metric macros e.g. `timer!` etc?

[the remark this answers is not in the document]
Comment (ayirr7) [reply rust_snuba/src/processors/generic_metrics.rs:299]: hm I think we should be able to, looks like the timer macros just uses `Timer` anyways

[the remark this answers is not in the document]
Comment (untitaker) [reply rust_snuba/src/processors/generic_metrics.rs:299]: yes

File: rust_snuba/src/processors/generic_metrics.rs (hunk 2 above)
Comment (ayirr7) [reply rust_snuba/src/processors/generic_metrics.rs]: done

**Review by untitaker — APPROVED** [review]

**Pull request merged**